pkg/queue/readiness/probe.go

```diff
@@ -39,6 +40,34 @@ type Probe struct {
 	*corev1.Probe
 	count       int32
 	pollTimeout time.Duration // To make tests not run for 10 seconds.
+
+	// Barrier sync to ensure only one probe is happening at the same time.
+	mu sync.RWMutex
+	gv *gateValue
+}
+
+// gateValue is a write-once boolean.
+type gateValue struct {
+	mu     sync.RWMutex
+	result bool
+}
+
+// newGV returns a gateValue which is ready to write.
+func newGV() *gateValue {
+	gv := &gateValue{}
+	gv.mu.Lock()
```

Comment:
Should we use a `chan struct{}` as a signal instead to avoid having a Lock here and an Unlock below which _looks_ dangerous 🤔

Reply:
Done :)
though I 💘  the original version :)